### ISSUE TITLE
Support new Sass module syntax at-rules

### ISSUE THREAD

**Issue by mirisuzanne** [kristerkari/stylelint-scss#366]
The new syntax ([now in beta](https://www.npmjs.com/package/sass/v/1.23.0-module.beta.1), planned for [release on Oct 1](https://github.com/sass/sass/blob/master/accepted/module-system.md#timeline)) replaces `@import` with `@forward` and `@use`. For now, I'm overriding stylelint-css with:

```yaml
rules:
  at-rule-no-unknown: false
  scss/at-rule-no-unknown:
    - true
    - ignoreAtRules:
      - use
      - forward
```

**Comment by kristerkari** [comment]
Does it require anyrthing else than just whitelisting those two at-rules?

**Comment by mirisuzanne** [comment]
Other syntax changes include namespaced variables, mixins, and functions - eg `module.$var`, `module.mixin()`, etc - but stylelint wasn't throwing errors for those, so I think this might be all? you can skim through [the spec](https://github.com/sass/sass/blob/master/accepted/module-system.md) and see if anything else stands out.

**Comment by mirisuzanne** [comment]
`@use` and  `@forward` also have new syntax internally, that doesn't seem to break the linter as far as I can tell…

```scss
@use 'my-module' as 'module' with (
  $base-color: $base-color,
  $secondary-color: color.scale($base-color, $lightness: -10%),
);
@forward 'another-module' show function-name, mixin-name, etc;
```

**Comment by nex3** [comment]
Even if the syntax parses, it's likely that you'll need to update a number of lints to be aware of the new module system semantics. You'll probably also want the Sass parser to fully parse at least `@use`, so that the linter can analyze expressions in the `with` clause.

**Comment by in-in** [comment]
I have a conflict of [operator-no-unspaced](https://github.com/kristerkari/stylelint-scss/blob/master/src/rules/operator-no-unspaced/README.md) rule and new Sass module syntax

`@use "styles/mixins" as *;`
![192821570633943](https://user-images.githubusercontent.com/8797432/66494513-7475ac00-eac0-11e9-9ec3-4b8375e6811f.png)

**Comment by tysongach** [comment]
Looks like https://github.com/kristerkari/stylelint-scss/pull/381 adds general support for `@use` - I left a comment on that PR about also including `@forward`.

**Comment by Guo-dalu** [comment]
@mirisuzanne  It seems that the `module.variable` syntax still lint errors in my project...
![image](https://user-images.githubusercontent.com/23010242/70110693-cd119180-168a-11ea-940d-9ecdeaaff2fb.png)

The lint npm modules have been all upated to latest, which is :
    "stylelint": "^12.0.0",
    "stylelint-config-sass-guidelines": "^6.2.0",
    "stylelint-scss": "^3.13.0"

And here is my .stylelintrc: 

```json
{
  "extends": [
    "stylelint-config-sass-guidelines"
  ],
  "plugins": ["stylelint-scss"],
  "rules": {
    "no-empty-source": null,
    "selector-max-id": 1,
    "order/properties-alphabetical-order": null,
    "max-nesting-depth": 3,
    "property-no-vendor-prefix": null,
    "scss/at-rule-no-unknown": null,
    "scss/at-import-partial-extension-blacklist": null,
    "selector-class-pattern":"^(?:(?:o|c|u|t|s|is|has|_|js|qa)-)?[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*(?:__[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*)?(?:--[a-zA-Z0-9]+(?:-[a-zA-Z0-9]+)*)?(?:\\[.+\\])?$"
  }
}

```

**Comment by tysongach** [comment]
I believe this issue can be closed now - support for `@use` and `@forward` was [shipped in v3.12.0](https://github.com/kristerkari/stylelint-scss/blob/master/CHANGELOG.md#3120) (yay!).
